Performance.md

```diff
@@ -492,6 +492,16 @@ Isolated file emit can be leveraged by using the following tools:
 * [ts-jest](https://kulshekhar.github.io/ts-jest/) can use be configured with the [`isolatedModules` flag set to `true`]isolatedModules: true(.
 * [ts-node](https://www.npmjs.com/package/ts-node) can detect [the `"transpileOnly"` option in the `"ts-node"` field of a `tsconfig.json`, and also has a `--transpile-only` flag](https://www.npmjs.com/package/ts-node#cli-and-programmatic-options).
 
+# Optimizing Editing Experience; Performance of `ts-server`
+
+`ts-server`'s performance characteristics will always be related to the
+performance of type-checking your entire project using `tsc`, so the other
+performance optimization guidance here also applies to improving the editing
+experience. As you type, the checker is completely starting from scratch, but it
+only requests information about what you're typing. This means that the editing
+experience can vary based on how much work TypeScript needs to do to check the
+type of what you are actively editing.
```

Comment:
I would like an additional sentence about errors:

1. the editor requests diagnostics in the background, a second or two after you stop typing.
2. the diagnostics are produced for all open files, not the entire project. So they will appear faster than a full `tsc` run, but slower than quick info, hover, etc. (As you mention already, error-reporting performance is closely related to `tsc` performance.)

Reply:
(2) is true for VS Code I believe, but not for VS

so you could say "most editors like VS Code"

Reply:
OK, I added some details about diagnostics.